### PR TITLE
[SDX-1256] No scroll when dropdown is opened v2

### PR DESCRIPTION
**Pull request by olga-andreeva**
* Choose dropdown's direction better

### CODE REVIEW

File: src/framework/base/dropdown/BaseDropdown.jsx [hunk 1]
```diff
@@ -164,17 +156,44 @@ export default class BaseDropdown extends Component {
...
+
+    const $optionList = $(this.refs.optionList);
+
+    const roomBelow =
+      (
+        $(window).height() - $input.offset().top - $input.parent().outerHeight()
+      ) + $(window).scrollTop();
+    const roomAbove = $input.offset().top - $(window).scrollTop();
+
+    const EXTRA_SPACE = 10;
```
Comment (mariacasancho) [inline src/framework/base/dropdown/BaseDropdown.jsx]: I think constants should be at the beginning

File: src/framework/base/dropdown/BaseDropdown.jsx [hunk 2]
```diff
@@ -164,17 +156,44 @@ export default class BaseDropdown extends Component {
...
+
+    if (roomBelow < $optionList.outerHeight()) {
```
Comment (mariacasancho) [inline src/framework/base/dropdown/BaseDropdown.jsx]: you can also do this "if" after getting the const roomBelow (before doing roomAbove) and return as you did with input.offset() so you save calculating roomAbove incase of false

File: src/framework/base/dropdown/BaseDropdown.jsx (hunk 1 above)
Comment (mohsinhijazee) [inline src/framework/base/dropdown/BaseDropdown.jsx]: Would be better to move it up.

Comment (olga-andreeva) [inline src/framework/base/dropdown/BaseDropdown.jsx]: I think this constant makes sense only in the context of `chooseOpeningDirection`. I don't know how to name it if it is in the constructor.

File: src/framework/base/dropdown/BaseDropdown.jsx (hunk 2 above)
Comment (olga-andreeva) [inline src/framework/base/dropdown/BaseDropdown.jsx]: If I understand you correctly, I should move `roomAbove` inside the `if` block.

**Pull request merged**